File: LeetCode/71. Simplify Path/Solution.cs
public class Solution
{
    public string SimplifyPath(string path)
    {
        var result = new StringBuilder();

        var dirs = path.Split("/");

        var stack = new Stack<string>();

        foreach (var dir in dirs)
            if (!String.IsNullOrWhiteSpace(dir) && dir != ".")
            {
                if (dir == "..")
                {
                    if (stack.Count() != 0)
                        stack.Pop();
                }
                else stack.Push(dir);
            }

        if (stack.Count() == 0) result.Append("/");
        while (stack.Count() != 0) result.Insert(0, "/" + stack.Pop());

        return result.ToString();
    }
}

File: LeetCode/71. Simplify Path/Solution.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function (path) {
  let dirs = path.split("/");
  let stack = [];

  dirs.forEach((dir) => {
    if (!isEmpty(dir) && dir != ".") {
      if (dir === "..") stack.pop();
      else stack.push(dir);
    }
  });

  return "/" + stack.join("/");
};

function isEmpty(value) {
  return (
    value == null || (typeof value === "string" && value.trim().length === 0)
  );
}
